### ISSUE TITLE
create domain ignoring NumberOfFilesLimited on WebServerLog

### ISSUE THREAD

**Issue by 53d117460ec63d70** [oracle/weblogic-deploy-tooling#303]
model extract:
```yaml
topology:
    Server:
        myserver:
            WebServer:
                WebServerLog:
                    FileCount: 5
                    BufferSizeKb: 0
                    NumberOfFilesLimited: true
```
produces config.xml extract:

```xml
      <web-server-log>
        <name>myserver</name>
        <file-count>5</file-count>
        <buffer-size-kb>0</buffer-size-kb>
      </web-server-log>
```
Attribute is confirmed as being ```false``` in WLST:
```python
wls:/mydomain/serverConfig> get('Servers/myserver/WebServer/myserver/WebServerLog/myserver/NumberOfFilesLimited')
0
```
Tested in WLS10.3.6. Setting the attribute works on wls12.1.3

**Comment by rakillen** [comment]
It appears that WLST offline has a default of true for NumberOfFilesLimited, and omits this attribute in the config.xml for that case.  In WLST online, the default appears to be false, and is not updated if absent in config.xml.  Checking if there is a patch for this.

**Comment by rakillen** [comment]
Note: This seems to be fixed in 12.1.3, offline omits the line in config.xml if the value is true, but online reads it correctly as true.

**Comment by ddsharpe** [comment]
For this project (WDT), it does not appear that we can do anything to workaround this issue in WLST.  Please file a support case with Oracle Support for this WLST issue and list the versions that you need to be fixed.  For this particular case, the fix for newer versions appears to already exists and the support team will need to do a backport of the existing fix to the versions you require.

**Comment by rakillen** [comment]
There is a work-around that may help, if it fits your process. If you run updateDomain (after domain create and start) with the online option and the same model, it should set the correct values for NumberOfFilesLimited and RotateLogOnStartup.  You could do this with a stripped-down model with just those elements if that's easier.

**Comment by ddsharpe** [comment]
This issue has been nagging at me.  So, I did some more testing and research.  This "bug" has not been fixed and still exists in the current WebLogic product.  WLST offline does not have a way to address the fact that an attribute has multiple defaults.  Therefore, it is not possible with WLST offline to set the value of NumberOfFilesLimited to true for a production mode domain.  For WDT, the only way to get this value to true in production mode would be to use WDT online.  Please file a bug against WebLogic Server for WLST offline to fix this bug.

@53d117460ec63d70 This is why you see the values you mention in your post:
The MBean attribute NumberOfFilesLimited has two defaults, "false" for production mode, and "true" for development mode.  WLST online determines the default value (if unset) based on whether the domain is running in production mode or development mode.  WLST offline only knows of one default, the development mode default which is "true".  Based on the data you posted, you tried to create a production mode domain and set the value for NumberOfFilesLimited to true.  WLST offline decided that "true" was the default, and leaves the config.xml attribute unset.  Therefore, the config.xml does not have the element for this attribute.  After you start the server in production mode, WLST online queries that attribute and determines the value is unset and returns the default for production mode, which is false.

**Comment by 53d117460ec63d70** [comment]
The suggested workaround https://github.com/oracle/weblogic-deploy-tooling/issues/303#issuecomment-476768240 gives the following error:
```
Caused by: oracle.weblogic.deploy.util.PyWLSTException: WLSDPLY-00072: JRF wlst method applyJRF not loaded with the executed WLST
```
I am running updateDomain.sh with the same arguments as createDomain.sh:
```
/home/rob/build/weblogic-deploy/bin/updateDomain.sh \
        -oracle_home /home/rob/oracle_home \
        -domain_home /home/rob/domain_home \
        -model_file /home/rob/build/mydomain.yml \
        -domain_type WLS
```

**Comment by rakillen** [comment]
The problem with applyJRF has also been fixed and merged to master, and will be in release 0.21.
